compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/WindowDecoration.desktop.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.window
+
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+
+/**
+ * Defines the options for window decoration.
+ */
+interface WindowDecoration {
+
+    /**
+     * Specifies that the default system decoration should be used.
+     */
+    object System : WindowDecoration
```

Comment:
If we decide to add customisation in the future (show/hide some buttons, hide titlebar), can we convert it to `class System`?

I think we can preserve binary compatibility, but source compatibility will be broken (we can live with some source incompatibility though).

Alternative names to consider: `Default`, `SystemDefault`. Choose what you think is good.

Reply:
Renamed to `SystemDefault`. If we want to add customizations we can add a class `System`.